packages/headless/src/features/commerce/search/search-actions.ts

```diff
@@ -75,13 +89,17 @@ export const executeSearch = createAsyncThunk<
 
     return {
       response: fetched.success,
-      queryExecuted: state.commerceQuery?.query,
+      originalQuery:
+        state.commerceQuery?.query !== undefined
```

Comment:
So there's no difference between `originalQuery` and `queryExecuted` if I'm reading this well ?

That does not seem right 🤔 
It should be `originalQuery=what the user actually submitted in the search box`, and `queryExecuted=what we receives from backend query correction if any`